### PR TITLE
Deployment fix

### LINKED ISSUE #138: Automatic deployment - never go back in time [3h]
Right now it's possible that automatic deployment reverts already deployed changes.
Deployment jobs might be created like this:
Job A
Job B

But executed in this order:
Job B
Job A

Then changes from Job B are reverted essentially.
To prevent that, we could store the timestamp of the Git commit in the artifact repository to ensure that we never deploy outdated code.

### PR DESCRIPTION
**Pull request by AronNovak**
#138

### CODE REVIEW

File: RoboFile.php [hunk 1]
```diff
@@ -463,17 +450,37 @@ public function deployPantheonSync(string $env = 'test', bool $do_deploy = TRUE)
...
+
+    $result = $this->taskExecStack()
+      ->stopOnFail()
       ->exec("terminus remote:drush $pantheon_terminus_environment -- uli")
       ->run()
       ->getExitCode();
+
     if ($result !== 0) {
-      throw new Exception('The site could not be fully updated at Pantheon. Try "ddev robo deploy:pantheon-install-env" manually.');
+      throw new Exception('Could not generate a login link. Try again manually or check earlier errors.');
```
Comment (amitaibu) [inline RoboFile.php:483]: I'd keep the `Try "ddev robo deploy:pantheon-install-env" manually`. It helped me on my last deploy to figure what's needed

**Review by amitaibu — APPROVED** [review]

File: RoboFile.php (hunk 1 above)
Comment (AronNovak) [reply RoboFile.php:483]: We have it above:
https://github.com/Gizra/drupal-starter/pull/156/files#diff-fed7dd361d1a8d696657705806d8e756ff49cc4659f2191ab0e91172a59be0d8R462

It's just more fine-grained now.

Comment (amitaibu) [reply RoboFile.php:483]: Sorry, I don't see it. Do we indicate now somewhere about `ddev robo deploy:pantheon-install-env` (i.e. re-install of env)?

Comment (AronNovak) [reply RoboFile.php:483]: image

Comment (AronNovak) [reply RoboFile.php:483]: Ah, I need to adapt it to the real command name, one sec.

Comment (amitaibu) [reply RoboFile.php:483]: :+1:

**Pull request merged**